synapse/handlers/federation_event.py

```diff
@@ -792,9 +792,41 @@ async def _process_pulled_events(
             ],
         )
 
+        # Check if we already any of these have these events.
+        # Note: we currently make a lookup in the database directly here rather than
+        # checking the event cache, due to:
+        # https://github.com/matrix-org/synapse/issues/13476
```

Comment:
We could add an additional comment about potentially adding a cached check back when "something like https://github.com/matrix-org/synapse/issues/13916 comes along and correctly invalidates the event cache."

Reply:
I went ahead and added [a comment](https://github.com/matrix-org/synapse/issues/13476) to the referenced bug instead, which should serve the same purpose and tie things together a bit better on GitHub.